File: examples_cplus/05_tanh_curve.cpp
#include "pysimdata_cplus/pysimdata_cplus.h"

#include <iostream>

int main() {
    pysimdata_cplus::TanhCurve gen;
    gen.generate();
    std::string dir = gen.save("output/tanh_curve", "data", false);
    std::cout << "Saved to: " << dir << std::endl;
    return 0;
}

File: examples/05_tanh_curve.py
"""TanhCurve: 双曲正切"""
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from pysimdata.function import TanhCurve

gen = TanhCurve(x_range=(-3, 3), num_points=100, y_shape=50)
data = gen.generate()
print(f"TanhCurve: {data.shape}")
gen.save_all("output/tanh_curve")